source/developers-guide/concept-cart-bundle/index.md

```diff
@@ -0,0 +1,424 @@
+```
+public function showDeliveriesAction()
+{
+    /** @var CartBundle\Infrastructure\StoreFrontFacade $facade */
+    $facade = $this->container->get('shopware_cart.store_front_facade');
+    $facade->add(
+        new CartBundle\Domain\LineItem\LineItem(
+            $number = 'SW10239',
+            CartBundle\Domain\Product\ProductProcessor::TYPE_PRODUCT,
+            $quantity = 50
+        )
+    );
+    $facade->add(
+        new CartBundle\Domain\LineItem\LineItem(
+            $number = 'SW10009',
+            CartBundle\Domain\Product\ProductProcessor::TYPE_PRODUCT,
+            $quantity = 25
+        )
+    );
+
+    /** @var CartBundle\Domain\Cart\CalculatedCart $cart */
+    $cart = $facade->getCalculated();
+
+    /** @var CartBundle\Domain\Delivery\Delivery $delivery */
+    foreach ($cart->getDeliveries() as $index => $delivery) {
+        echo "\n\n ---------- \n delivery #" . $index;
+
+        $price = $delivery->getPositions()->getPrices()->getTotalPrice();
+        echo "\n amount of delivery: " . $price->getPrice();
+        echo "\n tax amount of delivery: " . $price->getCalculatedTaxes()->getAmount();
+
+        echo "\n\n address: ";
+        echo ' ' . $delivery->getAddress()->getFirstname();
+        echo ' ' . $delivery->getAddress()->getLastname();
+        echo ' ' . $delivery->getAddress()->getStreet();
+        echo ' ' . $delivery->getAddress()->getZipcode();
+        echo ' ' . $delivery->getAddress()->getCity();
+
+        echo "\n delivery date: ";
+        echo $delivery->getDeliveryDate()->getEarliest()->format('Y-m-d');
+        echo ' - ';
+        echo $delivery->getDeliveryDate()->getLatest()->format('Y-m-d');
+
+        /** @var CartBundle\Domain\Delivery\DeliveryPosition $position */
+        foreach ($delivery->getPositions() as $i => $position) {
+            echo "\n\n position " . $i;
+            echo "\n quantity " . $position->getQuantity();
+            echo "\n price in delivery: " . $position->getPrice()->getPrice();
+        }
+
+    }
+}
+```
+
+## Architecture
+
+### Cart layers
+Inside the calculation process, the cart passes through different states. To provide a valid state for each service layer the this states are reflected in different classes:
+* `CartBundle\Domain\Cart\Cart`
+    * Defines which line items has to be calculated inside the process
+* `CartBundle\Domain\Cart\ProcessorCart`
+    * Defines which line items already calculated and which deliveries already generated 
+* `CartBundle\Domain\Cart\CalculatedCart`
+    * Contains a list of all calculated line items 
+    * Contains a collection of all generated deliveries
+    * Has a calculated price with summed taxes, tax rules and net or gross prices 
+
+### Processor concept 
+The following diagram shows the architecture behind the cart process for product calculation:   
+
+![image](img/img.001.png)
+
+The cart calculation are done over the `CartBundle\Domain\Cart\CartCalculator` class. 
```

Comment:
The cart calculation is done in the `CartBundle\Domain\Cart\CartCalculator` class.